spikeextractors/extractors/nwbextractors/nwbextractors.py

```diff
@@ -411,38 +416,46 @@ def add_electrode_groups(recording: se.RecordingExtractor, nwbfile=None, metadat
         if nwbfile is not None:
             assert isinstance(nwbfile, NWBFile), "'nwbfile' should be of type pynwb.NWBFile"
         if len(nwbfile.devices) == 0:
-            se.NwbRecordingExtractor.add_devices(recording, nwbfile)
-        defaults = dict(
-            name="Electrode Group",
-            description="no description",
-            location="unknown",
-            device_name="Device"
-        )
-        if metadata is None or 'ElectrodeGroup' not in metadata['Ecephys']:
-            metadata = dict(
-                Ecephys=dict(
-                    ElectrodeGroup=[defaults]
-                )
+            se.NwbRecordingExtractor.add_devices(recording=recording, nwbfile=nwbfile, metadata=metadata)
+
+        if metadata is None:
+            metadata = dict()
+
+        if 'Ecephys' not in metadata:
+            metadata['Ecephys'] = dict()
+
+        defaults = [
+            dict(
+                name=str(group_id),
+                description="no description",
+                location="unknown",
+                device=[i.name for i in nwbfile.devices.values()][0]
```

Comment:
I'm OK with this but you should note that (as apparent in the new tests) it is a rare behavior that has to be forced via the exclusion of an entire `ElectrodeGroup` key in the metadata, which is inconsistent with most defaulting use via `get_nwb_metadata`.